addons/l10n_in/report/account_invoice_report.py

```diff
@@ -230,9 +230,16 @@ def _sub_select(self):
                 CASE WHEN tag_rep_ln.account_tax_report_line_id IN
                     (SELECT res_id FROM ir_model_data WHERE module='l10n_in' AND name='tax_report_line_sgst') OR at.l10n_in_reverse_charge = True
                     THEN NULL
-                    ELSE (CASE WHEN aml.tax_base_amount <> 0 THEN aml.tax_base_amount ELSE NULL END)
+                    ELSE (CASE WHEN aml.tax_base_amount <> 0 THEN aml.tax_base_amount * (CASE WHEN aml.balance < 0 THEN -1 ELSE 1 END) ELSE NULL END)
                     END AS price_total,
                 (CASE WHEN aj.type = 'sale' AND (am.type IS NULL OR am.type != 'out_refund') THEN -1 ELSE 1 END) AS amount_sign,
```

Comment:
@jva-odoo No possibility to change amount_sign directly instead of an extra b2cs_refund_sign?  What is the difference with v12 that it gives errors now?

Reply:
If I directly change amount_sign then it also gives a negative sign in the CDNR section and this is wrong.
So I do this using b2cs_refund_sign.

CDNR = Credit/Debit Notes for registered

Reply:
@jco-odoo customer is frustrated now plz check

Reply:
@jva-odoo Then you could have said something sooner.  Does it still pass the tests?  Does it solve all the problems?  Who has been testing this?  (Sorry, but these are significant changes and I have not got the time to figure everything out at once)  Why do you change the sign when the amount is bigger at a certain point?  I suppose it is for the large companies as you describe in your commit message but it is weird this is just changing the sign.

Reply:
@jco-odoo 
This is because there is a separate section for a bigger amount(B2CL) and lover amount(B2CS).

In bigger amount(B2CL) detail per invoices and the amount is not negative because there is one collum where we defined its invoice or credit note.

In lower amount(B2CS) detail as per Indian states so here we need to minus(-) a refund.

This is tested by Indian CA (NOD).